lua/entities/kleiner/cl_init.lua

```diff
@@ -0,0 +1,142 @@
+-- This file is subject to copyright - contact swampservers@gmail.com for more information.
+-- INSTALL: CINEMA
+
+if(SERVER)then return end
+include("shared.lua")
+language.Add("kleiner", "Dr. Isaac Kleiner")
+
+function ENT:Initialize()
+		local effectdata = EffectData()
+	effectdata:SetOrigin( self:GetPos() )
+	effectdata:SetEntity( self )
+	util.Effect( "propspawn", effectdata )
+end
+ 
+
+local spkr = Material("voice/icntlk_pl")
+hook.Add("CreateClientsideRagdoll","KleinerRagdollsFix",function(ent,rag)
+	if(IsValid(ent) and ent:GetClass() == "kleiner")then
+		if(ent.GetBased)then rag.wasbased = ent:GetBased() end
+		function rag:GetPlayerColor()
+			if(self.wasbased)then
+				return Vector(0.8,0,0.05)
+			end
+			return Vector(0.23,0.35,0.41)
+		end
+		
+		
+		if(IsValid(rag))then rag:SetSubMaterial(5,ent:GetSubMaterial(5,"")) end
+		timer.Create(ent:EntIndex().."ragcleaner",10,1,function()
+			if(IsValid(rag))then
+				rag:Remove()
+			end
+		end)
+	end
+end)
+
+
+
+function ENT:GetStareTarget()
+	local target = self:GetTarget()
+
+	if(IsValid(target))then 
+		return target:EyePos()
+	end
+	return self:EyePos() + self:GetAngles():Forward()*600
+end
+
+
+function ENT:Think()
+local lod = self:GetPos():Distance(EyePos())
+
+		if(lod < 400)then
+		local targetpos = self:GetStareTarget() or Vector()
+		local angle = (targetpos - self:EyePos()):Angle()
+		local pos,ang = WorldToLocal(targetpos,angle,self:EyePos(),self:GetAngles())
+		self:SetPoseParameter("head_yaw",ang.yaw/2)
+		self:SetPoseParameter("head_pitch",ang.pitch/4)
+			if(lod < 200)then
+				self:SetEyeTarget(targetpos)
+			end
+		end	
+end
+
+function ENT:Draw()
+
+	self:DrawModel()
+	
+	if(self:GetTalking() > CurTime())then
+		local bone = self:GetBonePosition(self:LookupBone("ValveBiped.Bip01_Head1") or 0)
+		render.SetMaterial( spkr )
+		render.DrawSprite( bone + Vector(0,0,16), 12, 12, color_white)
+	end
+end
+
+function ENT:GetPlayerColor()
+	if(self:GetBased())then
+	return Vector(0.8,0,0.05)
```

Comment:
can these colors (since they are copied in the ragdoll) be put in a single location?